### ISSUE TITLE
Add return value in non-void function

### ISSUE THREAD

**Issue by heikojansen** [Leont/perlio-utf8_strict#3]
Had a problem today installing this module on openSUSE Tumbleweed (x64, Perl 5.30.1):

````
utf8_strict.xs: In function ‘validate’:
utf8_strict.xs:164:1: error: control reaches end of non-void function [-Werror=return-type]
  164 | }
      | ^
````
(plus a few non-fatal warnings).
Then found that there's a RPM package for the dist which installed successfully.
The package contains a small patch for the problem show above:

https://build.opensuse.org/package/view_file/openSUSE:Factory/perl-PerlIO-utf8_strict/no-return-in-nonvoid-function.patch?expand=1

Maybe that's worth including upstream directly in your code?

**Comment by Leont** [comment]
Both report functions throw an exception, so the function not returning isn't really a problem. Arguably they should be marked as such so that the compiler also realizes this.

**Comment by voegelas** [comment]
Can you release a new version of PerlIO::utf8_strict that fixes this issue? The prototypes in my pull request #5 are enclosed by #ifdef. Nothing changes for old Perl versions that do not provide the definitions that mark functions as noreturn.

I sometimes build Task::Kensho from scratch on several platforms for testing purposes. PerlIO::utf8_strict is the only module that fails to build on openSUSE Tumbleweed.

**Comment by Leont** [comment]
Should be fixed in 0.009
